File: Projeto/src/main/java/br/uece/aps/classes/livro/Categoria.java
package br.uece.aps.classes.livro;

import java.util.ArrayList;
import java.util.List;

public enum Categoria {
	AUTOBIOGRAFIA("autobiografia"),
	BIOGRAFIA("biografia"),
	FANTASIA("fantasia"),
	FICCAOCIENTIFICA("ficção-cientifica"),
	HORROR("horror"),
	ROMANCE("romance"),
	AUTOAJUDA("autoajuda"),
	RECEITA("receita");
	
	private final String categoria;
	Categoria(String categoria) {
		this.categoria = categoria;
	}
	
	public String getCategoria() {
		return this.categoria;
	}	
	
	public static String[] getListaCategoria() {
		List<String> list = new ArrayList<>();
		for(Categoria it : Categoria.values()) {
			list.add(it.getCategoria());
		}
		return list.toArray(new String[0]);
	}
}

File: livraria_uece/lib/classes/livro/categoria.dart
class Categoria {
  String _id;
  String _categoria;

  Categoria(this._id, this._categoria);

  String get categoria => _categoria;

  set categoria(String value) {
    _categoria = value;
  }

  String get id => _id;

  set id(String value) {
    _id = value;
  }

  String toString(){
    return "Categoria "+_id.toString()+' '+_categoria.toString();
  }
}
